staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -50,5 +79,177 @@ func NewREST(resource schema.GroupResource, listKind schema.GroupVersionKind, st
 	if err := store.CompleteWithOptions(options); err != nil {
 		panic(err) // TODO: Propagate error up
 	}
-	return &REST{store}
+
+	statusStore := *store
+	statusStore.UpdateStrategy = NewStatusStrategy(strategy)
+	return &REST{store}, &StatusREST{store: &statusStore}
+}
+
+// Implement CategoriesProvider
+var _ rest.CategoriesProvider = &REST{}
+
+// Categories implements the CategoriesProvider interface. Returns a list of categories a resource is part of.
+func (r *REST) Categories() []string {
+	return []string{"all"}
+}
+
+// StatusREST implements the REST endpoint for changing the status of a CustomResource
+type StatusREST struct {
+	store *genericregistry.Store
+}
+
+func (r *StatusREST) New() runtime.Object {
+	return &unstructured.Unstructured{}
+}
+
+// Get retrieves the object from the storage. It is required to support Patch.
+func (r *StatusREST) Get(ctx genericapirequest.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	return r.store.Get(ctx, name, options)
+}
+
+// Update alters the status subset of an object.
+func (r *StatusREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc) (runtime.Object, bool, error) {
+	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation)
+}
+
+type ScaleREST struct {
+	registry      Registry
+	scaleJSONPath map[string]string
```

Comment:
why a map?

Reply:
would make all of the paths explicit here

Reply:
To have something like this:

```
scaleJSONPath["specReplicasPath"] = crd.Spec.SubResources.Scale.SpecReplicasPath
scaleJSONPath["statusReplicasPath"] = crd.Spec.SubResources.Scale.StatusReplicasPath
scaleJSONPath["labelSelectorPath"] = crd.Spec.SubResources.Scale.LabelSelectorPath
```

Reply:
> would make all of the paths explicit here
:+1:

Reply:
3 string fields would do as well?

Reply:
yes :+1: